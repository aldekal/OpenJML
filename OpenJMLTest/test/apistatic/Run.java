import java.io.*;
import org.openjml.IAPI;

public class Run {

 public static void main(String... args) {
   PrintWriter pw = new PrintWriter(System.out);
   int ex = -1;
   try {
     IAPI api = IAPI.make();
     ex = api.execute("--esc", "A.java");
     System.out.println("EXIT: " + ex);
     ex = api.execute(pw, "--esc", "B.java");
     System.out.println("EXIT: " + ex);
     ex = api.execute(System.out , "--esc", "B.java");
     System.out.println("EXIT: " + ex);
   } finally {
     pw.close();
   }
 }
}
