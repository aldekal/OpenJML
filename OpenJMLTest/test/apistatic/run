#! /usr/bin/env bash

F1="$BASH_SOURCE[0]"
F2=$(dirname "$F1")
cd "$F2"

./runx |& diff - RUN.out
if [ $? -ne 0 ]; then echo test/apistatic FAILED ; fi
