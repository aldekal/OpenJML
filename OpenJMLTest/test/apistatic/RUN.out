A.java:6: verify: Show statement expression x has value false
//@ show x;
         ^
A.java:7: verify: The prover cannot establish an assertion (Assert) in method main
//@ assert false;
    ^
2 verification failures
EXIT: 6
B.java:1: error: class A is public, should be declared in a file named A.java
public class A {
       ^
1 error
EXIT: 1
B.java:1: error: class A is public, should be declared in a file named A.java
public class A {
       ^
1 error
